eng/pipelines/templates/stages/vmr-final-join.yml

```diff
@@ -0,0 +1,61 @@
+parameters:
+# Branch of the VMR to use (to push to for internal builds)
+- name: vmrBranch
+  type: string
+  default: $(Build.SourceBranch)
+
+- name: pool_Windows
+  type: object
+  default:
+    name: $(defaultPoolName)
+    image: $(poolImage_Windows)
+    demands: ImageOverride -equals $(poolImage_Windows)
+    os: windows
+
+- name: mainVertical
+  type: string
+  default: Windows_x64
+    
+stages:
+- stage: VMR_Final_Join
+  displayName: VMR Final Join
+  dependsOn: VMR_Vertical_Build
+  variables:
+  - group: Publish-Build-Assets
+  - template: ../variables/vmr-build.yml
+    parameters:
+      vmrBranch: ${{ parameters.vmrBranch }}
+
+  jobs:
+  - job: VMR_Final_Join
+    pool: ${{ parameters.pool_Windows }}
+
+    templateContext:
+      outputs:
+      - output: pipelineArtifact
+        path: $(Build.ArtifactStagingDirectory)/artifacts
+        artifact: JoinedArtifacts
+        condition: succeededOrFailed()
+
+    steps:
+    - checkout: self
+
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        artifactName: 'VerticalManifests'
+        targetPath: $(Build.ArtifactStagingDirectory)/VerticalManifests
+
+    - powershell: |
+        . $(Build.SourcesDirectory)/eng/common/tools.ps1
+        InitializeToolset
+        ./.dotnet/dotnet restore $(Build.SourcesDirectory)/eng/tools/join-verticals.proj
+        MSbuild $(Build.SourcesDirectory)/eng/tools/join-verticals.proj `
```

Comment:
Why don't you just invoke the build script and pass `DotNetBuildPass=final` in and condition build.proj appropriately? That's the behavior that I intended when writing the join point documented. I think it's better to just have a single build entry point and not directly call into eng/common/tools.ps1 from YML.

Reply:
Good point, I updated the code to reflect this, not 100% sure if this is how you imagined it